orm/versioning_bucket.go

```diff
@@ -107,18 +107,28 @@ func (b VersioningBucket) GetVersion(db weave.ReadOnlyKVStore, ref VersionedIDRe
 	return b.Get(db, key)
 }
 
-// Create stores a the given data. It assigns an ID and initial version number to the object instance and returns the
+// Create stores the given data. It assigns an ID and initial version number to the object instance and returns the
 // VersionedIDRef which won't be nil on success.
 func (b VersioningBucket) Create(db weave.KVStore, data versionedData) (*VersionedIDRef, error) {
-	if data.GetVersion() != 0 {
-		return nil, errors.Wrap(errors.ErrInput, "version is set on create")
-	}
-	data.SetVersion(1)
 	newID, err := b.idGen.NextVal(db, data)
 	if err != nil {
 		return nil, err
 	}
-	idRef := VersionedIDRef{ID: newID, Version: data.GetVersion()}
+	return b.create(db, newID, data)
+}
+
+// CreateWithID stores the given data. It accepts an ID and assigns an initial version number to the object instance
+// and returns the VersionedIDRef which won't be nil on success.
+func (b VersioningBucket) CreateWithID(db weave.KVStore, id []byte, data versionedData) (*VersionedIDRef, error) {
```

Comment:
pp: I can see why you want the ID as an argument but the `WithVersioning(b IDGenBucket) VersioningBucket` constructor requires an IDGenBucket which is fully ignored by this method now. 

Why not drop that dependency and the other `Create` method and always require an ID argument? The only other place beside tests that uses the method is the electorate Bucket.

Reply:
Main reason is that `CreateWithID` is basically used to cater for the Address edge-case, in other instances it's nice to have an autogenerated id.

Since they are referencing the same sequence - I'd say it should be fine to have both.
There might be a slick(er) solution for this actually, but that'd require using interfaces instead of structs and a bigger change.